system/mp/test/mptest.cpp

```diff
@@ -701,6 +704,411 @@ void MPAlltoAll(IGroup *group, ICommunicator *mpicomm, size32_t buffsize=0, unsi
+            if (std::find(tgtRanks.begin(), tgtRanks.end(), sender) == tgtRanks.end())
+                throwStringExceptionV(0, "Received reply from rank(%u) this stream did not send to", (unsigned)sender);
+            msg.read(ack);
+            msg.clear();
+            return true;
+        }
+        void verifyAck(unsigned receivedHash, unsigned expectedHash)
+        {
+            if (receivedHash != expectedHash)
+                throwStringExceptionV(0, "Checksums mismatch: %u sent vs %u received", expectedHash, receivedHash);
+        }
+
+    public:
+        CSendStream(ICommunicator *_comm, rank_t _myRank, unsigned _grpSize, mptag_t _mpTag, unsigned __int64 _totalSendSize, size32_t _msgSize, bool _async)
+            : threaded("CSendStream", this), comm(_comm), myRank(_myRank), grpSize(_grpSize), mpTag(_mpTag), totalSendSize(_totalSendSize), msgSize(_msgSize), ackThread(*this, _mpTag), async(_async)
+        {
+            if (1 == grpSize) // group only contains self, so target self
+                tgtRanks.push_back(0);
+            else
+            {
+                unsigned pc=25; // %
+
+                // add 'pc'% targets, starting from myRank+1
+                unsigned num = grpSize*pc/100;
+                if (0 == num)
+                    num = 1;
+                unsigned step = grpSize / num;
+                unsigned t = myRank+1;
+                if (t == grpSize)
+                    t = 0;
+                while (true)
+                {
+                    tgtRanks.push_back(t);
+                    --num;
+                    if (0 == num)
+                        break;
+                    t += step;
+                    if (t >= grpSize)
+                        t -= grpSize;
+                }
+            }
+
+            auto iter = tgtRanks.begin();
+            while (true)
+            {
+                tgtRanksStr.append(*iter);
+                iter++;
+                if (iter == tgtRanks.end())
+                    break;
+                tgtRanksStr.append(",");
+            }
+            replyTag = createReplyTag();
+            if (async)
+                ackThread.start();
+            threaded.start();
+        }
+        ~CSendStream()
+        {
+            threaded.join();
+        }
+        bool waitResult(StringBuffer &resultMessage)
+        {
+            threaded.join();
+            return passed;
+        }
+        const char *queryTgtRanks() { return tgtRanksStr; }
+        virtual void threadmain() override
+        {
+            passed = false;
+            try
+            {
+                CMessageBuffer msg, recvMsg;
+                msg.setReplyTag(replyTag);
+                void *data = msg.reserveTruncate(msgSize);
+                unsigned hash = (unsigned)mpTag;
+
+                VStringBuffer logMsg("NxN: mpTag=%u, dstRank(s) [%s]", (unsigned)mpTag, tgtRanksStr.str());
+                PROGLOG("%s", logMsg.str());
```

Comment:
PROGLOG() or PrintLog() ?  Does it matter ?  Most of the other code uses PrintLog().

Reply:
doesn't really matter, but PrintLog is legacy I think (but is mapped to JLOG functionality), so should be using PROGLOG or similar.